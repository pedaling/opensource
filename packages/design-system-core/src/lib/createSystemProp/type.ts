import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
