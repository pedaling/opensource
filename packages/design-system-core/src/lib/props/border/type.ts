import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
