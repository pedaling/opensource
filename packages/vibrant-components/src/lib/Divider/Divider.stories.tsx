import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Divider } from './Divider';

export default {
  title: 'Divider',
  component: Divider,
  args: {
    direction: 'horizontal',
    margin: 'md',
    thickness: 1,
  },
} as ComponentMeta<typeof Divider>;

export const Basic: ComponentStory<typeof Divider> = props => <Divider {...props} />;
