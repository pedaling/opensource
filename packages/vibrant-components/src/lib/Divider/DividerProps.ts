import type { ResponsiveValue } from '@vibrant-ui/core';
import { propVariant, withVariation } from '@vibrant-ui/core';

export type DividerProps = {
  direction: 'horizontal' | 'vertical';
  kind?: ResponsiveValue<'dashed' | 'default' | 'thick'>;
  margin?: ResponsiveValue<'lg' | 'md' | 'none'>;
};

export const withDividerVariation = withVariation<DividerProps>()(
  propVariant({
    props: [
      {
        name: 'margin',
        responsive: true,
        default: 'none',
      },
    ],
    variants: {
      none: {
        spacing: 0,
      },
      md: {
        spacing: 16,
      },
      lg: {
        spacing: 24,
      },
    },
  }),
  propVariant({
    props: [
      {
        name: 'kind',
        responsive: true,
        keep: true,
        default: 'default',
      },
    ],
    variants: {
      dashed: {
        borderColor: 'outline1',
        size: 1,
        borderStyle: 'dashed',
      },
      thick: {
        borderColor: 'outline2',
        size: 12,
        borderStyle: 'solid',
      },
      default: {
        borderColor: 'outline1',
        size: 1,
        borderStyle: 'solid',
      },
    } as const,
  }),
  propVariant({
    props: [
      {
        name: 'direction',
      },
      {
        name: 'size',
      },
      {
        name: 'spacing',
        responsive: true,
      },
    ],
    variants: ({ direction, size, spacing }) => {
      if (direction === 'horizontal') {
        return {
          width: '100%',
          height: 0,
          borderTopWidth: size,
          my: spacing,
        };
      }

      return {
        width: 0,
        borderRightWidth: size,
        height: '100%',
        mx: spacing,
      };
    },
  })
);
