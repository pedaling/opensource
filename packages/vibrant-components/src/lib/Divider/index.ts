export { Divider } from './Divider';
export type { DividerProps } from './DividerProps';
