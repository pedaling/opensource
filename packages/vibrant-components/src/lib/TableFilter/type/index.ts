export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
