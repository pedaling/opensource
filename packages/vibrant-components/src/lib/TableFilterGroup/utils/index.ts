import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
