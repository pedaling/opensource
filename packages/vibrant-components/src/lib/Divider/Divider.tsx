import { Box } from '@vibrant-ui/core';
import { withDividerVariation } from './DividerProps';

export const Divider = withDividerVariation(({ ...props }) => <Box flex="none" {...props} />);
