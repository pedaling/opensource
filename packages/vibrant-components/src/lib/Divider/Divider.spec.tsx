import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Divider } from './Divider';

describe('<Divider />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;
  let element: HTMLElement;

  describe('when Divider with margin rendered', () => {
    beforeEach(() => {
      renderer = render(<Divider data-testid="divider" direction="horizontal" margin="md" kind="default" />);

      element = renderer.getByTestId('divider');
    });

    it('my is 16', () => {
      expect(element).toHaveStyleRule('margin-top', '16px');

      expect(element).toHaveStyleRule('margin-bottom', '16px');
    });

    it('width is 100%', () => {
      expect(element).toHaveStyleRule('width', '100%');
    });

    it('border-top-width is 1px', () => {
      expect(element).toHaveStyleRule('border-top-width', '1px');
    });

    it('match snapshot', () => {
      expect(element).toMatchSnapshot();
    });
  });
});
