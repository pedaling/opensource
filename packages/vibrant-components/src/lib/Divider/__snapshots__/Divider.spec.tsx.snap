// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> when Divider with margin rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  border-width: 0;
  border-color: rgba(0, 0, 0, 0.15);
  border-style: solid;
  width: 100%;
  height: 0;
  border-top-width: 1px;
  margin-top: 16px;
  margin-bottom: 16px;
}

<div
  class="emotion-0"
  data-testid="divider"
  kind="default"
/>
`;
